1.如何实现一个可以中断的方法   在方法里面执行下面的语句即可 
```java
   if (Thread.interrupted())
        throw new InterruptedException();   //这里说明 if判定为true 方法将不会继续往下执行 会抛出异常
```

2.
```java
//代码1  若一段代码前有异常抛出，并且这个异常没有被捕获，这段代码将产生编译时错误「无法访问的语句」
public static void test() throws Exception  {

    throw new Exception("参数越界"); 
    System.out.println("异常后"); //编译错误，「无法访问的语句」
}


//代码2 若一段代码前有异常抛出，并且这个异常被try...catch所捕获，若此时catch语句中没有抛出新的异常，则这段代码能够被执行，否则，同第1条
try{
    throw new Exception("参数越界"); 
}catch(Exception e) {
    e.printStackTrace();
}
System.out.println("异常后");//可以执行

 
//代码3  若在一个条件语句中抛出异常，则程序能被编译，但后面的语句不会被执行
if(true) {
    throw new Exception("参数越界"); 
}
System.out.println("异常后"); //抛出异常，不会执行

```


3.java 的异常体系
   运行时异常： RuntimeException类及其子类的异常  NullPointerException   IndexOutOfBoundsException
      运行时异常在程序中可以不进行处理(捕获或者向上抛出) 这个时候虚拟机会进行处理 虚拟机处理就是会进行停机 停止线程 打印异常
      同样可以在程序中进行处理 将程序将根据具体情况具体分析
   非运行异常：是RuntimeException以外的异常，类型上都属于Exception类及其子类，是受检异常。
      非运行时异常必须进行处理（捕获或向上抛出），如果不处理，程序将出现编译错误 比如InterruptedException异常



4.fail-fast事件












